### PR TITLE
Feature: 1.21 + black fishs + default fish color

### PR DESCRIPTION
**Pull request by AntoninHuaut**

### CODE REVIEW

**Comment by davepusey** [comment]
Should that be BucketVariantTag rather than custom_model_data ?

File: src/utils/variant.ts [hunk 1]
```diff
@@ -1,19 +1,6 @@
...
-    'kob',
-    'snooper',
-    'spotty',
-    'stripey',
-    'sunstreak',
-];
+export const types: string[] = ['betty', 'blockfish', 'brinely', 'clayfish', 'dasher', 'flopper', 'glitter', 'kob', 'snooper', 'spotty', 'stripey', 'sunstreak'];
 export const typesMapping: Record<string, [number, number]> = {
-    // first least byte / second-least byte : https://minecraft.fandom.com/wiki/Tropical_Fish (Entity data)
+    // first least byte / second-least byte : https://minecraft.fandom.com/wiki/Tropical_Fish#Entity_data
```
Comment (zer0p1us) [inline src/utils/variant.ts:3]: Probably best to point to the independent wiki [wiki](https://minecraft.wiki/w/Tropical_Fish#Entity_data), the fandom one is abandoned and unmaintained

**Comment by AntoninHuaut** [comment]
> Should that be BucketVariantTag rather than custom_model_data ?

Yes, but same issue with /give @p tropical_fish_bucket[minecraft:bucket_entity_data={BucketVariantTag:37671009}] :(

File: src/utils/variant.ts (hunk 1 above)
Comment (AntoninHuaut) [reply src/utils/variant.ts]: Yes thanks, I updated it

**Pull request merged**